Update code to node 10 ~ 12

mjschutz opened the pull request:
I update the code to build/run on node 12 (and 10), did some tests, but need more tests to do. The code can be changed to have macros to allow building for multiple versions of node (like I did for BooleanValue).

Works on NW.js too, I'm working on a system for the place I work and I will use this module (on NW.js).

mjschutz commented:
I will work on the macros for node version checking.

xdenser commented:
Build failing 
https://travis-ci.org/xdenser/node-firebird-libfbclient/jobs/602102371

mjschutz commented:
I know, node 12 dropped some codes and then the old one's will not work anymore, the macros will be need to check the node version (I will make another commit to fix that).

mjschutz commented:
It's not, I did the changes already and it's building on node +6, I replaced the val->ToObject and other conversions with Nan::To, and other Nan functions, the only problem was String::Concat, I changed it to use a macro to add Isolate::GetCurrent() if it's node +10.

Nan::MakeCallback is deprecated and I replaced it in favor of AsyncResource::runInAsyncScope and seens to be working.

mariuz commented:
also node 6 is deprecated https://nodejs.org/en/about/releases/
